engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -102,6 +103,16 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
         throw new SpelCompilationException(node, e)
     }
 
+    def typeIndexer(e: Indexer, stack: List[TypingResultWithContext]): NodeTypingResult = {
+      stack match {
+        case TypingResultWithContext(TypedClass(clazz, param :: Nil), _) :: Nil if clazz.isAssignableFrom(classOf[java.util.List[_]]) => valid(param)
+        case TypingResultWithContext(TypedClass(clazz, keyParam :: valueParam :: Nil), _):: Nil if clazz.isAssignableFrom(classOf[java.util.Map[_, _]]) => valid(valueParam)
+        case TypingResultWithContext(d: TypedDict, _) :: Nil => dictTyper.typeDictValue(d, e).map(toResult)
+        case TypingResultWithContext(TypedTaggedValue(underlying, _), _) :: Nil => typeIndexer(e, TypingResultWithContext(underlying.objType) :: Nil)
+        case _ => if(dynamicPropertyAccessAllowed) valid(Unknown) else invalid("Dynamic property access is not allowed")
```

Comment:
I think we need to handle one more case - Union. This probably will amount to having sequence on all Union types

Reply:
I just appended my match expression but I'm not sure about my solution. Does it fit your idea?